doc/policy_based_metering/PBM-HLD.md

```diff
@@ -0,0 +1,352 @@
+        "OP": "SET"
+    }
+}
+```
+#### CLI Config Commands
+```bash
+# Add/Remove/Update policer --> no change
+config policer add <name> --rate <rate> --burst <burst>
+config policer remove <name>
+config policer update <name> --rate <rate> --burst <burst>
+
+# 'config load' command to configudre the new ACL
+config load acl_cfg.json
+```
+#### CLI Show Commands
+```bash
+# Show existing ACL tables --> no change
+show acl table [OPTIONS] [TABLE_NAME]
+
+# Show existing ACL rules --> no change
+show acl rule [OPTIONS] [TABLE_NAME] [RULE_ID]
+```
+Example:
+```
+admin@sonic:~$ show acl table
+Name         Type    Binding    Description          Stage    * Extended Actions    Status
+-----------  ------  ---------  -------------------  -------    -----------------   --------
+Table_ID_#1  L3      Ethernet2  Permit some traffic  Ingress    POLICER             Inactive
+                     Ethernet4
+                     Ethernet7
+
+
+admin@sonic:~$ show acl rule
+Table         Rule          Priority      Actions                    Match
+--------      ------------  ----------    -------------------------  ----------------------------
+Table_ID_#1   RULE_5        9995          REDIRECT: Ethernet8        IP_PROTOCOL: 126
+                                        * POLICER: policer2
+
+Table_ID_#1   RULE_6        9994          FORWARD                    L4_SRC_PORT: 179
+                                        * POLICER: policer3
+
+SNMP_ACL      DEFAULT_RULE  23            DROP                       ETHER_TYPE: 2048
+```
+
+#### YANG Model Enhancements
+
+Add 'policer' to sonic-yang-models/yang-templates/sonic-types.yang.j2:
+```yang
+typedef packet_action {
+	type enumeration {
+		enum DROP;
+        enum ACCEPT;
+        enum FORWARD;
+        enum REDIRECT;
+        enum DO_NOT_NAT;
+        enum MIRROR;
+      * enum POLICER;
+    }
+}
+```
+---
+### Warmboot and Fastboot Design Impact
+N/A
+
+---
+### Memory Consumption
+N/A
+
+---
+### Restrictions/Limitations
+
+- Policers must be supported
+- Policer cannot be deleted when referenced by ACL rule.
+- PRE-INGRESS stage isn't supported.
+---
+### Testing Requirements/Design
+#### Unit Test Cases
+- Test ACL-Orchagent and Policer-Orchagent logic for correct processing.
+#### System Test Cases
+- Ensure correct packet marking based on policer configurations.
+- Warm/Fast reboot tests to check the changes do not break these features.
+#### CLI Level Tests
+- Verify command run successfully with valid parameter enable/disable.
+-  Verify command abort with invalid policer parameter.
+- Verify command output.
+#### DB validation
+- Verify CONFIG DB is correctly updated.
+---
+### Open/Action Items
+- Are there specific ASIC platforms with limitations for this feature?
+- ACL-Loader utility uses the an external python library (openconfig_acl) that rely on pre-defined, structured models. In order to support the new 'policer_action' per rule the openconfig_acl schema need to be changed.
```

Comment:
why is this open? effort? what is the the side effect?

Reply:
Theoretically, there is an option to be a contributor for openconfig_acl (.py) (https://openconfig.net/about/contribute/) - but...